bigbluebutton-html5/imports/ui/components/user-list/component.jsx

```diff
@@ -30,6 +31,128 @@ class UserList extends Component {
     this.state = {
       compact: this.props.compact,
     };
+
+    this.rovingIndex = this.rovingIndex.bind(this);
+    this.focusList = this.focusList.bind(this);
+    this.focusListItem = this.focusListItem.bind(this);
+    this.counter = -1;
+  }
+
+  focusList(activeElement, list) {
+    activeElement.tabIndex = -1;
+    this.counter = 0;
+    list.tabIndex = 0;
+    list.focus();
+  }
+
+  focusListItem(active, direction, element, count) {
+
+    function select() {
+      element.tabIndex = 0;
+      element.focus();
+    }
+
+    active.tabIndex = -1;
+
+    switch (direction) {
+      case 'down':
+        element.childNodes[this.counter].tabIndex = 0;
+        element.childNodes[this.counter].focus();
+        this.counter++;
+        break;
+      case 'up':
+        this.counter--;
+        element.childNodes[this.counter].tabIndex = 0;
+        element.childNodes[this.counter].focus();
+        break;
+      case 'upLoopUp':
+      case 'upLoopDown':
+        this.counter = count - 1;
+        select();
+        break;
+      case 'downLoopDown':
+        this.counter = -1;
+        select();
+        break;
+      case 'downLoopUp':
+        this.counter = 1;
+        select();
+        break;
+    }
+  }
+
+  rovingIndex(...Args) {
+    const { users, openChats } = this.props;
+
+    let active = document.activeElement;
+    let list;
+    let items;
+    let count;
+
+    switch (Args[1]) {
+      case 'users':
+        list = this._usersList;
+        items = this._userItems;
+        count = users.length;
+        break;
+      case 'messages':
+        list = this._msgsList;
+        items = this._msgItems;
+        count = openChats.length;
+        break;
+    }
+
+    if (Args[0].keyCode === KEY_CODES.ESCAPE
+      || this.counter === -1
+      || this.counter > count) {
+      this.focusList(active, list);
+    }
+
+    if (Args[0].keyCode === KEY_CODES.ENTER
+        || Args[0].keyCode === KEY_CODES.ARROW_RIGHT
+        || Args[0].keyCode === KEY_CODES.ARROW_LEFT) {
+      active.firstChild.click();
+    }
+
+    if (Args[0].keyCode === KEY_CODES.ARROW_DOWN) {
+      if (this.counter < count) {
+        this.focusListItem(active, 'down', items);
+      }else if (this.counter === count) {
+        this.focusListItem(active, 'downLoopDown', list);
+      }else if (this.counter === 0) {
+        this.focusListItem(active, 'downLoopUp', list);
+      }
+    }
+
+    if (Args[0].keyCode === KEY_CODES.ARROW_UP) {
+      if (this.counter < count && this.counter !== 0) {
+        this.focusListItem(active, 'up', items);
+      }else if (this.counter === 0) {
+        this.focusListItem(active, 'upLoopUp', list, count);
+      }else if (this.counter === count) {
+        this.focusListItem(active, 'upLoopDown', list, count);
+      }
+    }
+  }
+
+  componentDidMount() {
+    let _this = this;
```

Comment:
Why are you doing this? You can use arrow functions to prevent losing the context/scope.